### PR TITLE
RNMT-1053 Truncate excess logs

### PR DESCRIPTION
**Pull request by EiyuuZack**

### CODE REVIEW

**Review by bgracio — APPROVED** [review]

File: puree/src/main/java/com/cookpad/puree/outputs/PureeBufferedOutput.java [hunk 1]
```diff
@@ -26,6 +28,12 @@ public PureeBufferedOutput() {
     @Override
     public void initialize(PureeLogger logger) {
         super.initialize(logger);
+
+        if(storage.count() > MAX_LOG_CAPACITY) {
```
Comment (luissilvaos) [inline puree/src/main/java/com/cookpad/puree/outputs/PureeBufferedOutput.java]: I don't like this. We are adding OutSystems logic into the Puree Framework.
This logic must be added to our Logger plugin. We must create a class on our plugin that extends this class and add this logic into it.
Let's try to keep the Puree Framework clean as possible.

File: puree/src/main/java/com/cookpad/puree/storage/PureeSQLiteStorage.java [hunk 2]
```diff
@@ -127,7 +130,7 @@ public void delete(Records records) {
 
     @Override
     public void truncateBufferedLogs(int maxRecords) {
-        int recordSize = getRecordCount();
+        int recordSize = count();
```
Comment (luissilvaos) [inline puree/src/main/java/com/cookpad/puree/storage/PureeSQLiteStorage.java:133]: Good catch!

File: puree/src/main/java/com/cookpad/puree/storage/PureeStorage.java [hunk 3]
```diff
@@ -4,12 +4,13 @@
...
 
-    public void insert(String type, JsonObject jsonLog);
-    public Records select(String type, int logsPerRequest);
-    public Records selectAll();
-    public void delete(Records records);
-    public void truncateBufferedLogs(int maxRecords);
-    public void clear();
-    public boolean lock();
-    public void unlock();
+    void insert(String type, JsonObject jsonLog);
```
Comment (luissilvaos) [inline puree/src/main/java/com/cookpad/puree/storage/PureeStorage.java:7]: Good catch!

**Review by luissilvaos — CHANGES_REQUESTED** [review]
See my comments bellow.

**Review by luissilvaos — APPROVED** [review]
👍

**Pull request merged**